.github/workflows/continuous-integration.yml

```diff
@@ -54,7 +54,15 @@ jobs:
 
     - name: Run all checks
       run: |
-        nix flake check --print-build-logs
+        # this is just `nix flake check --print-build-logs`
+        # everything else is to buffer the output for speed
+        (
+          set +e
+          nix flake check --print-build-logs >out 2>&1
+          errcode=$?
+          cat out
+          exit $errcode
+        )
```

Comment:
Is the subshell necessary here? I would have naively expected GitHub Actions to run each step in its own shell.